app/models/game.rb

```diff
@@ -3,46 +3,38 @@ class Game < ApplicationRecord
     Bishop.create(game_id: id, user_id: user_id, position_x: 5, position_y: 7, color: "black", name: "black bishop 2")
-    # white bishops
-    Bishop.create(game_id: id, user_id: opponent, position_x: 2, position_y: 0, color: "white", name: "white bishop 1")
-    Bishop.create(game_id: id, user_id: opponent, position_x: 5, position_y: 0, color: "white", name: "white bishop 2")
-    # KNIGHTS
-    # black knights
     Knight.create(game_id: id, user_id: user_id, position_x: 1, position_y: 7, color: "black", name: "black knight 1")
     Knight.create(game_id: id, user_id: user_id, position_x: 6, position_y: 7, color: "black", name: "black knight 2")
-    # white knights
-    Knight.create(game_id: id, user_id: opponent, position_x: 1, position_y: 0, color: "white", name: "white knight 1")
-    Knight.create(game_id: id, user_id: opponent, position_x: 6, position_y: 0, color: "white", name: "white knight 2")
-    # rooks
-    # black rooks
     Rook.create(game_id: id, user_id: user_id, position_x: 0, position_y: 7, color: "black", name: "black rook 1")
     Rook.create(game_id: id, user_id: user_id, position_x: 7, position_y: 7, color: "black", name: "black rook 2")
-    # white rooks
-    Rook.create(game_id: id, user_id: opponent, position_x: 0, position_y: 0, color: "white", name: "white rook 1")
-    Rook.create(game_id: id, user_id: opponent, position_x: 7, position_y: 0, color: "white", name: "white rook 2")
-    # PAWNS
-    # populate black pawns
     black_pawn_x = 0
     black_pawn_number = 1
     8.times do
       Pawn.create(game_id: id, user_id: user_id, position_x: black_pawn_x, position_y: 6, color: "black", name: "black pawn #{black_pawn_number}")
       black_pawn_x += 1
       black_pawn_number += 1
     end
-    # populate white pawns
+  end
+
+  # upon game update, populate the white pieces for the opponent user who joined the game
+  def populate_white_pieces!
+    Queen.create(game_id: id, user_id: opponent, position_x: 3, position_y: 0, color: "white", name: "white queen")
+    King.create(game_id: id, user_id: opponent, position_x: 4, position_y: 0, color: "white", name: "white king")
+    Bishop.create(game_id: id, user_id: opponent, position_x: 2, position_y: 0, color: "white", name: "white bishop 1")
+    Bishop.create(game_id: id, user_id: opponent, position_x: 5, position_y: 0, color: "white", name: "white bishop 2")
+    Knight.create(game_id: id, user_id: opponent, position_x: 1, position_y: 0, color: "white", name: "white knight 1")
+    Knight.create(game_id: id, user_id: opponent, position_x: 6, position_y: 0, color: "white", name: "white knight 2")
+    Rook.create(game_id: id, user_id: opponent, position_x: 0, position_y: 0, color: "white", name: "white rook 1")
+    Rook.create(game_id: id, user_id: opponent, position_x: 7, position_y: 0, color: "white", name: "white rook 2")
     white_pawn_x = 0
```

Comment:
So just to make sure I'm understanding this correctly, the board is initially populated with just the user's pieces (black) and then later the board is populated with the opponent's pieces (white) after they've joined the game? What are your thoughts on just leaving the `opponent_id` of the pieces `nil` until an opponent is added? That way the entire board is displayed at the beginning of the game. Let me know what you think of this.

Reply:
Yes, that's how it works. My understanding is that leaving opponent as `nil` is exactly what our current `games#create` action does, and that leaves half the board unpopulated.

Reply:
What we were doing before this PR was creating a dummy opponent before the creation of a game. That only worked for one game. If you try to create another game then the white pieces are missing because there is only one opponent in the entire db. To fix that, I removed the creation of the dummy opponent. The problem with that is every piece needs a user, so only the owner of the game is getting pieces created. By simply removing the relationship between users and pieces, this allows us to create the white pieces without a user associated with them.

Reply:
I did this in one of my recent PR's so it shouldn't be a problem anymore.

Reply:
But as we apply the logic for an opponent to join a game, all we should have to do is apply the opponent's id to each of the white pieces for that game.

Reply:
This way all 32 pieces appear on the board when the board is rendered as would happen when playing other online chess games.

Reply:
I removed the dummy opponent function from my local branch a while ago since, as I mentioned in slack, it was causing errors for me when I was testing my code. I'm not sure what you have added to fix this, but the code I have here has been working for me both on this branch and (as I just wrote in Slack) on the most recent version of the master branch.